### PR TITLE
Committee renewal

### PR DESCRIPTION
**Pull request by troels-im**
This PR contains an initial version of committee renewal.

### CODE REVIEW

**Review by ejfitzgerald — COMMENTED** [review]
Seems like a good starting point. Would be worth checking with @n-hutton and @jinmannwong I have not spun up on consensus at the moment since my head is still very much in the network stack

File: apps/beacon/main.cpp [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+};
+
+int main()
+{
+  constexpr uint16_t number_of_nodes    = 16;
+  constexpr uint16_t cabinet_size       = 4;
+  constexpr uint16_t number_of_cabinets = number_of_nodes / cabinet_size;
+
+  // Initialising the BLS library
+  crypto::bls::Init();
```
Comment (n-hutton) [inline apps/beacon/main.cpp:106]: Seems like the codebase indicates to use ```BLSInitialiser``` here

File: libs/vm/src/generator.cpp [hunk 2]
```diff
@@ -416,10 +416,10 @@ void Generator::HandleWhileStatement(IRBlockNodePtr const &block_node)
   function_->AddInstruction(jf_instruction);
   AddLineNumber(condition_node->line, jf_pc);
 
-  uint16_t const while_block_start_pc = uint16_t(function_->instructions.size());
+  uint16_t const while_round_start_pc = uint16_t(function_->instructions.size());
```
Comment (n-hutton) [inline libs/vm/src/generator.cpp]: is this supposed to be here?

File: libs/core/include/core/serializers/binary_interface.hpp [hunk 3]
```diff
@@ -114,6 +114,9 @@ class BinaryConstructorInterface
           std::string("Cannot create container type with more than 1 << 32 elements"));
     }
 
+    // Allocating needed mempry
+    serializer_.Allocate(count);
```
Comment (n-hutton) [inline libs/core/include/core/serializers/binary_interface.hpp:118]: unrelated?

Comment (troels-im) [reply libs/core/include/core/serializers/binary_interface.hpp:118]: It's related in that only BLS uses the binary serialiser and this solves a problem with it.

File: libs/vm/src/generator.cpp (hunk 2 above)
Comment (troels-im) [reply libs/vm/src/generator.cpp]: Good catch.

File: libs/beacon/include/beacon/beacon_setup_service.hpp [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+  struct DeliveryDetails
+  {
+    bool             was_delivered{false};
+    service::Promise response{nullptr};
+  };
+
+  BeaconSetupService()                           = delete;
+  BeaconSetupService(BeaconSetupService const &) = delete;
+  BeaconSetupService(BeaconSetupService &&)      = delete;
+  explicit BeaconSetupService(Endpoint &endpoint, Identity identity);
```
Comment (n-hutton) [inline libs/beacon/include/beacon/beacon_setup_service.hpp]: no need for explicit here
